FFmpeg 6.0 deprecated ‘AVCodecContext::frame_number’

Building DDNet 16.8 having FFmpeg 6.0 installed issues the this deprecation warning. Looks like the frame_number deprecation is documented in [6.0 docs](https://ffmpeg.org/doxygen/trunk/structAVCodecContext.html#a1606ae31b14af5a2940b94c99264c0fc), which was not in [5.1 docs](https://ffmpeg.org/doxygen/5.1/structAVCodecContext.html#a9e5a25a530d01c04491216c368a1a04a).

Output:
```
[154/268] Building CXX object CMakeFiles/DDNet.dir/src/engine/client/video.cpp.o
/build/aur/ddnet/src/DDNet-16.8/src/engine/client/video.cpp: In member function ‘void CVideo::RunVideoThread(size_t, size_t)’:
/build/aur/ddnet/src/DDNet-16.8/src/engine/client/video.cpp:555:115: warning: ‘AVCodecContext::frame_number’ is deprecated [-Wdeprecated-declarations]
  555 |                                         m_VideoStream.m_vpFrames[ThreadIndex]->pts = (int64_t)m_VideoStream.pEnc->frame_number;
      |                                                                                                                   ^~~~~~~~~~~~
In file included from /build/aur/ddnet/src/DDNet-16.8/src/engine/client/video.h:7,
                 from /build/aur/ddnet/src/DDNet-16.8/src/engine/client/video.cpp:20:
/usr/include/libavcodec/avcodec.h:1076:9: note: declared here
 1076 |     int frame_number;
      |         ^~~~~~~~~~~~
/build/aur/ddnet/src/DDNet-16.8/src/engine/client/video.cpp:555:115: warning: ‘AVCodecContext::frame_number’ is deprecated [-Wdeprecated-declarations]
  555 |                                         m_VideoStream.m_vpFrames[ThreadIndex]->pts = (int64_t)m_VideoStream.pEnc->frame_number;
      |                                                                                                                   ^~~~~~~~~~~~
/usr/include/libavcodec/avcodec.h:1076:9: note: declared here
 1076 |     int frame_number;
      |         ^~~~~~~~~~~~
/build/aur/ddnet/src/DDNet-16.8/src/engine/client/video.cpp:555:115: warning: ‘AVCodecContext::frame_number’ is deprecated [-Wdeprecated-declarations]
  555 |                                         m_VideoStream.m_vpFrames[ThreadIndex]->pts = (int64_t)m_VideoStream.pEnc->frame_number;
      |                                                                                                                   ^~~~~~~~~~~~
/usr/include/libavcodec/avcodec.h:1076:9: note: declared here
 1076 |     int frame_number;
      |         ^~~~~~~~~~~~
At global scope:
cc1plus: note: unrecognized command-line option ‘-Wno-nullability-completeness’ may have been intended to silence earlier diagnostics
```

--- a/src/engine/client/video.cpp
+++ b/src/engine/client/video.cpp
@@ -28,6 +28,12 @@ using namespace std::chrono_literals;
 
 #define STREAM_PIX_FMT AV_PIX_FMT_YUV420P /* default pix_fmt */
 
+#if LIBAVCODEC_VERSION_MAJOR >= 60
+#define FRAME_NUM frame_num
+#else
+#define FRAME_NUM frame_number
+#endif
+
 const size_t FORMAT_GL_NCHANNELS = 4;
 LOCK g_WriteLock = 0;
 
@@ -325,7 +331,7 @@ void CVideo::NextVideoFrameThread()
 				}
 			}
 
-			//dbg_msg("video_recorder", "vframe: %d", m_VideoStream.pEnc->frame_number);
+			//dbg_msg("video_recorder", "vframe: %d", m_VideoStream.pEnc->FRAME_NUM);
 
 			// after reading the graphic libraries' frame buffer, go threaded
 			{
@@ -371,7 +377,7 @@ void CVideo::NextAudioFrameTimeline(ISoundMixFunc Mix)
 {
 	if(m_Recording && m_HasAudio)
 	{
-		//if(m_VideoStream.pEnc->frame_number * (double)m_AudioStream.pEnc->sample_rate / m_FPS >= (double)m_AudioStream.pEnc->frame_number * m_AudioStream.pEnc->frame_size)
+		//if(m_VideoStream.pEnc->FRAME_NUM * (double)m_AudioStream.pEnc->sample_rate / m_FPS >= (double)m_AudioStream.pEnc->FRAME_NUM * m_AudioStream.pEnc->frame_size)
 		double SamplesPerFrame = (double)m_AudioStream.pEnc->sample_rate / m_FPS;
 		while(m_AudioStream.m_SamplesFrameCount >= m_AudioStream.m_SamplesCount)
 		{
@@ -552,7 +558,7 @@ void CVideo::RunVideoThread(size_t ParentThreadIndex, size_t ThreadIndex)
 				std::unique_lock<std::mutex> LockVideo(pThreadData->m_VideoFillMutex);
 				{
 					CLockScope ls(g_WriteLock);
-					m_VideoStream.m_vpFrames[ThreadIndex]->pts = (int64_t)m_VideoStream.pEnc->frame_number;
+					m_VideoStream.m_vpFrames[ThreadIndex]->pts = (int64_t)m_VideoStream.pEnc->FRAME_NUM;
 					WriteFrame(&m_VideoStream, ThreadIndex);
 				}
 

